src/pytest_split/plugin.py

```diff
@@ -134,4 +134,37 @@ def _calculate_suite_start_and_end_idx(splits, group, items, stored_durations):
     if not end_idx:
         end_idx = len(items)
 
+    # Modify start_idx and end_idx so they match start/end of an ipynb file
+    if _is_ipy_notebook(item_node_ids[start_idx]):
+        start_idx = _get_boundary_idx_for_ipynb(item_node_ids, start_idx, "start")
+    if _is_ipy_notebook(item_node_ids[end_idx - 1]):  # since Python indexing is [a, b)
+        end_idx = _get_boundary_idx_for_ipynb(item_node_ids, end_idx, "end")
```

Comment:
Here, I will first inspect whether the node_id associted with `start_idx` and `end_idx` are IPython notebooks or not, and if so, I modify the indices so a single IPython notebook is not broken into different test groups.